src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -178,7 +178,37 @@ bool DockPageView::isDockOpen(const QString& dockName) const
 
 void DockPageView::toggleDock(const QString& dockName)
 {
-    setDockOpen(dockName, !isDockOpen(dockName));
+    DockBase* dock = dockByName(dockName);
+    if (dock == nullptr) {
+        return;
+    }
+
+    bool shouldOpen = true;
+
+    DockPanelView* panel = dynamic_cast<DockPanelView*>(dock);
+    if (panel) {
+        if (panel->isOpen()) {
+            if (panel->floating()) {
+                shouldOpen = !panel->hasFocus();
+            } else {
+                DockPanelView* panelParent = findCurrentPanelForTab(panel);
+                if (panelParent) {
+                    const int currentTabIndex = panelParent->currentTabIndex();
+                    const int tabIndexOfPanel = panelParent->tabIndexOfPanel(panel);
+                    Q_ASSERT(tabIndexOfPanel >= 0);
+                    shouldOpen = !panel->hasFocus() || currentTabIndex != tabIndexOfPanel;
+                } else {
+                    shouldOpen = !panel->hasFocus();
+                }
+            }
+        } else {
+            shouldOpen = true;
+        }
+    } else {
+        shouldOpen = !panel->hasFocus() || !panel->isOpen();
+    }
```

Comment:
At a first glance, this looks a bit strange to me... so we have 
```c++
    if (panel {
        ...
    } else {
        // !panel 
        shouldOpen = !panel->hasFocus() || !panel->isOpen(); // still using panel?
    }
```

Or am I misreading it?

Maybe you wanted one of these:
```c++
    if (panel && panel->isOpen()) {
        ...
    } else {
        shouldOpen = true;
    }
```

```c++
    if (!panel) {
        return;
    }   

    if (panel->isOpen()) {
        ...
    } else {
        shouldOpen = true;
    }
```

Reply:
Yes, I carelessly renamed all my usages of dock to panel, but shouldn't have changed the panel == nullptr case. I'll fix that soon to be more like:
```
    if (panel {
        ...
    } else {
        shouldOpen = !dock->hasFocus() || !dock->isOpen();
    }
```

Also, I seem to be in a weird state with git right now, where following [this git workflow](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/git-workflow#Summary) has gotten me into a never ending conflict loop when I try to rebase. And just when I think it's resolved, I try again and it still thinks there are conflicts even after they are resolved. 

I may have to nuke it from orbit if I can't figure out how to get through this cleanly. Any tips?

Reply:
@cbjeukendrup This has been updated, and I figured out my git issues (merging when I should have been only rebasing updates).

Reply:
Looks good now indeed!